src/jaxsim/rbda/contacts/relaxed_rigid.py

```diff
@@ -0,0 +1,356 @@
+                HashedNumpyArray(self.damping),
+                HashedNumpyArray(self.mu),
+                HashedNumpyArray(self.max_iterations),
+                HashedNumpyArray(self.tolerance),
+            )
+        )
+
+    def __eq__(self, other: RelaxedRigidContactsParams) -> bool:
+        return hash(self) == hash(other)
+
+    @classmethod
+    def build(
+        cls,
+        time_constant: jtp.Float | None = None,
+        damping_coefficient: jtp.Float | None = None,
+        d_min: jtp.Float | None = None,
+        d_max: jtp.Float | None = None,
+        width: jtp.Float | None = None,
+        midpoint: jtp.Float | None = None,
+        power: jtp.Float | None = None,
+        stiffness: jtp.Float | None = None,
+        damping: jtp.Float | None = None,
+        mu: jtp.Float | None = None,
+    ) -> RelaxedRigidContactsParams:
+        """Create a `RelaxedRigidContactsParams` instance"""
+
+        return cls(
+            **{
+                field: locals().get(field) or cls.__dataclass_fields__[field].default
+                for field in cls.__dataclass_fields__
+                if field != "__mutability__"
+            }
+        )
+
+    def valid(self) -> bool:
+        return bool(
+            jnp.all(self.time_constant >= 0.0)
+            and jnp.all(self.damping_coefficient > 0.0)
+            and jnp.all(self.d_min >= 0.0)
+            and jnp.all(self.d_max <= 1.0)
+            and jnp.all(self.d_min <= self.d_max)
+            and jnp.all(self.width >= 0.0)
+            and jnp.all(self.midpoint >= 0.0)
+            and jnp.all(self.power >= 0.0)
+            and jnp.all(self.mu >= 0.0)
+            and jnp.all(self.max_iterations > 0)
+            and jnp.all(self.tolerance > 0.0)
+        )
+
+
+@jax_dataclasses.pytree_dataclass
+class RelaxedRigidContactsState(ContactsState):
+    """Class storing the state of the relaxed rigid contacts model."""
+
+    def __eq__(self, other: RelaxedRigidContactsState) -> bool:
+        return hash(self) == hash(other)
+
+    @staticmethod
+    def build() -> RelaxedRigidContactsState:
+        """Create a `RelaxedRigidContactsState` instance"""
+
+        return RelaxedRigidContactsState()
+
+    @staticmethod
+    def zero(model: js.model.JaxSimModel) -> RelaxedRigidContactsState:
+        """Build a zero `RelaxedRigidContactsState` instance from a `JaxSimModel`."""
+        return RelaxedRigidContactsState.build()
+
+    def valid(self, model: js.model.JaxSimModel) -> bool:
+        return True
+
+
+@jax_dataclasses.pytree_dataclass
+class RelaxedRigidContacts(ContactModel):
+    """Rigid contacts model."""
+
+    parameters: RelaxedRigidContactsParams = dataclasses.field(
+        default_factory=RelaxedRigidContactsParams
+    )
+
+    terrain: jax_dataclasses.Static[Terrain] = dataclasses.field(
+        default_factory=FlatTerrain
+    )
+
+    def compute_contact_forces(
+        self,
+        position: jtp.Vector,
+        velocity: jtp.Vector,
```

Comment:
```suggestion
        position: jtp.VectorLike,
        velocity: jtp.VectorLike,
```

Reply:
Done in https://github.com/ami-iit/jaxsim/pull/223/commits/89843345142c7d73b88be31940d376cf947bcf38